packages/cadl-ranch-specs/http/azure/resource-manager/resources/toplevel.tsp

```diff
@@ -235,4 +235,51 @@ interface TopLevel {
       ```
     """)
   actionSync is ArmResourceActionNoContentSync<TopLevelTrackedResource, NotificationDetails>;
+
+  @scenario
+  @scenarioDoc("""
+      Resource POST operation.
+      Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/providers/Azure.ResourceManager.Resources/checkNameAvailability
+      Expected query parameter: api-version=2023-12-01-preview
+      Expected request body:
+      ```json
+      {
+        "name": "checkName",
+        "type": "global"
+      }
+      ```
+      Expected response body:
+      ```json
+      {
+        "nameAvailable": true,
+        "reason": "AlreadyExists",
```

Comment:
`"nameAvailable": true` should not mean it `"AlreadyExists"`?

Reply:
Fixed in the version.